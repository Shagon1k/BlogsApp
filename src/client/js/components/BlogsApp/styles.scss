.blogsAppContainer {
	width: 40%;
	padding: 20px 65px;
	margin: 0 auto;
	background: #50626C;
	display: flex;
	flex-flow: column wrap;
	align-items: center;
	justify-content: space-between;
}