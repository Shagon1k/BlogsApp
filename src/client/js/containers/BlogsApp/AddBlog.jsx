import { connect } from 'react-redux';
import { addBlog } from '../../actions';
import AddBlogModal from '../../components/BlogsApp/modals/AddBlogModal/index.jsx';

const mapStateToProps = (state, ownProps) => ({
	onClose: () => {ownProps.history.push('/blogs')}
})

const mapDispatchToProps = dispatch => ({
		onConfirm: (blog) => {dispatch(addBlog(blog))}
})

const AddBlog = connect(
	mapStateToProps,
	mapDispatchToProps
)(AddBlogModal)

export default AddBlog;