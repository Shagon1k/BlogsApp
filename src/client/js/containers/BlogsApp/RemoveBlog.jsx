import { connect } from 'react-redux';
import { sendRequestDeleteBlog } from '../../actions';
import RemoveBlogModal from '../../components/BlogsApp/modals/RemoveBlogModal/index.jsx';

const mapStateToProps = (state, ownProps) => {
	const blogObj = state.blogs.find((i) => i.id === ownProps.match.params.id);
	return {
		blogTitle: blogObj.blog.title
	}
}

const mapDispatchToProps = (dispatch, ownProps) => ({
	onConfirm: () => {
		dispatch(sendRequestDeleteBlog(ownProps.match.params.id));
	}
})

const RemoveBlog = connect(
	mapStateToProps,
	mapDispatchToProps
)(RemoveBlogModal)

export default RemoveBlog;