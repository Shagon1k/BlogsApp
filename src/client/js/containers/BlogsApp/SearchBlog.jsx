import { connect } from 'react-redux';
import { setSearchValue, setSearchParam } from '../../actions';
import SearchBlog from '../../components/BlogsApp/SearchBlog/index.jsx';

const mapStateToProps = state => ({
	search: state.search
})

const mapDispatchToProps = dispatch => {
	return {
		onValueChange: value => {
			dispatch(setSearchValue(value))
		},
		onParamChange: param => {
			dispatch(setSearchParam(param))
		}
	}
}

const SearchBlogContainer = connect(
	mapStateToProps,
	mapDispatchToProps
)(SearchBlog)

export default SearchBlogContainer;