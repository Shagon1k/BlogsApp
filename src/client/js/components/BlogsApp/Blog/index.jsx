import React from 'react';
import { Link } from 'react-router-dom';
import './styles.scss';

const Blog = ({blog, blogId}) => (
	<li className="blogBox">
		<dl className="blogInfo">
			<dt> Title: </dt>
			<dd> {blog.title} </dd>
			<dt> Author: </dt>
			<dd> {blog.author} </dd>
			<dt> Message: </dt>
			<dd> {blog.message}</dd>
		</dl>
		<Link to={`/blogs/remove/${blogId}`} className="removeBlog"> X </Link>
		<Link className="blogDetailsLink" to={`/blog/${blogId}`}> Blog details </Link>
	</li>
)

export default Blog;