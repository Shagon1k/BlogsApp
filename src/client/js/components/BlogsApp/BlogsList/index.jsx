import React from 'react';
import Blog from '../Blog/index.jsx';
import './styles.scss';


class BlogsList extends React.Component {
	constructor(props) {
		super(props);
	}

	componentDidMount() {
		this.props.initBlogs();
	}

	render() {
		return (
			<ul className="blogList">
				{
					this.props.blogs.map((i) => <Blog key={i.id} blog={i.blog} blogId={i.id} />)
				}
			</ul>
		);
	}
}

export default BlogsList;