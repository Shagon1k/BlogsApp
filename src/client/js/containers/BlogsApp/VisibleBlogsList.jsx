import { connect } from 'react-redux';
import { initBlogs } from '../../actions';
import BlogsList from '../../components/BlogsApp/BlogsList/index.jsx';

const getVisibleBlogs = (blogs, search) => {
	let searchParam;
	if (search.searchParam === 'BY_TITLE') {
		searchParam = 'title';
	} else if (search.searchParam === 'BY_AUTHOR') {
		searchParam = 'author';
	}
	
	return blogs.filter(i => i.blog[searchParam].includes(search.searchValue))
}

const mapStateToProps = state => ({
	blogs: getVisibleBlogs(state.blogs, state.search)
})

const mapDispatchToProps = dispatch => ({
	initBlogs: () => {
		dispatch(initBlogs());
	}
})

const VisibleBlogsList = connect(
	mapStateToProps,
	mapDispatchToProps
)(BlogsList)

export default VisibleBlogsList;