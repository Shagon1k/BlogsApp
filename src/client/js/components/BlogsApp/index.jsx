import React from 'react';
import AddBlogBtn from './AddBlogButton/index.jsx';
import SearchBlog from '../../containers/BlogsApp/SearchBlog.jsx';
import VisibleBlogsList from '../../containers/BlogsApp/VisibleBlogsList.jsx';
import AddBlog from '../../containers/BlogsApp/AddBlog.jsx';
import RemoveBlog from '../../containers/BlogsApp/RemoveBlog.jsx';
import { Route } from 'react-router-dom';
import './styles.scss';

const BlogsApp = () => (
	<div className="blogsAppContainer">
		<AddBlogBtn />
		<SearchBlog />
		<VisibleBlogsList />
		<Route path='/blogs/add' component={AddBlog} />
		<Route path='/blogs/remove/:id' component={RemoveBlog} />
	</div>
)

export default BlogsApp;