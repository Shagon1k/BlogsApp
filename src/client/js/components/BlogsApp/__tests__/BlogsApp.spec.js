import React from 'react';
import BlogsApp from '../index.jsx';

describe('BlogsApp component', () => {
	it('renders BlogsApp correctly', () => {
		const wrapper = shallow(<BlogsApp/>);
		expect(wrapper).toMatchSnapshot();
	});
})