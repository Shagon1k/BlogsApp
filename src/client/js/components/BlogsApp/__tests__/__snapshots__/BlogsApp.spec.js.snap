// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogsApp component renders BlogsApp correctly 1`] = `
<div
  className="blogsAppContainer"
>
  <Link
    className="lnk homePageLnk"
    replace={false}
    to="/"
  >
    Home Page
  </Link>
  <AddBlogBtn />
  <Connect(SearchBlog) />
  <Connect(BlogsList) />
  <Route
    component={[Function]}
    path="/blogs/add"
  />
  <Route
    component={[Function]}
    path="/blogs/remove/:id"
  />
</div>
`;
